cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -143,27 +221,39 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
     try {
       EndPoint endPoint = endPoint(request);
       if (endPoint != null) {
-        switch (endPoint) {
-          case BOOTSTRAP:
-            bootstrap(request, response);
-            break;
-          case TRAIN:
-            train(request, response);
-            break;
-          case LOAD:
-            getClusterLoad(request, response);
-            break;
-          case PARTITION_LOAD:
-            getPartitionLoad(request, response);
-            break;
-          case PROPOSALS:
-            getProposals(request, response);
-            break;
-          case STATE:
-            getState(request, response);
-            break;
-          default:
-            throw new UserRequestException("Invalid URL for GET");
+        Set<String> validParamNames = VALID_ENDPOINT_PARAM_NAMES.get(endPoint);
+        if (validParamNames != null && !validParamNames.containsAll(request.getParameterMap().keySet())) {
+          // User request specifies parameters that are not a subset of the valid parameters.
+          String errorResp;
+          if (wantJSON(request)) {
+            errorResp = "{\"error\": \"Invalid endpoint parameters in " + endPoint + " get request.\"}";
```

Comment:
We probably want to return the set of invalid parameters to the users. Otherwise it is difficult for the users to know what was going wrong.